src/common/utxobased/engine/makeUtxoEngine.ts

```diff
@@ -523,53 +539,67 @@ export async function makeUtxoEngine(
       const tmpMetadata = await makeMetadata(tmpConfig)
       const tmpProcessor = await makeProcessor(tmpConfig)
       const tmpWalletTools = makeUtxoWalletTools({
-        publicKey: dummyNumbWalletInfo.keys.publicKey,
-        wifKeys: privateKeys,
+        publicKey: tmpWalletInfo.keys.publicKey,
         coin: coinInfo.name
       })
 
+      // Max spend after imported wallet finishes sync
       tmpEmitter.on(EngineEvent.ADDRESSES_CHECKED, async (ratio: number) => {
         if (ratio === 1) {
-          await tmpState.stop()
-
-          const utxos = (await processor.fetchUtxos({ utxoIds: [] })) as IUTXO[]
-          if (utxos === null || utxos.length < 1) {
-            throw new Error('Private key has no funds')
+          try {
+            // Delay to ensure that UTXOs are processed.
+            // This is a bug in the engine state; addresses are considered
+            // processed before the UTXOs are saved.
+            // TODO: Remove delay when fixed.
+            await delay(1000)
```

Comment:
This should not even be as a temporary solution.
Lucky for us, the way to fix it is super simple, all you need to do is:

1. Add this `await args.onAddressChecked()` at line 1135 in 'makeUtxoEngineState.ts'
2. Change line 126 in  'makeUtxoEngineState.ts' to be: `processedCount += 0.5`

Reply:
Clever solution. Though I will want to add a pretty verbose comment explaining why processedCount is only incremented by 0.5.